trackpy/feature.py

```diff
@@ -552,30 +563,38 @@ def locate(raw_image, diameter, minmass=100., maxsize=None, separation=None,
                 # To avoid degrading performance, assume gamut is zero to one.
                 # Have you ever encountered an image of unnormalized floats?
                 raw_image = 1 - raw_image
-        image = bandpass(raw_image, noise_size, smoothing_size, threshold)
+        image, black_level, noise = bandpass(raw_image, noise_size,
```

Comment:
I have not read the bandpass code yet, but I am guessing the last kwarg turn on/off the return of `black_level` and `noise` which (as commented on `scale_to_gamut`) is really bad practice.  In this case I would add a `bandpass_with_details` function and leave the api of `bandpass` as-is.

In terms of  implementation I am not sure of the best way (as I have not looked at the code yet).